Dockerfile

```diff
@@ -4,7 +4,7 @@ WORKDIR /opt/nodejs
 ENV NODE_VERSION v8.9.4
 
 RUN groupadd -r nodejs && \
-    useradd -r -g nodejs nodejs -d /app && \
+    useradd -r -u 1000 -g nodejs nodejs -d /app && \
```

Comment:
It might be worth choosing a UID far above 1000 to prevent clashes should a user be added to the base image. (As 1000 is the default first non-system user.)

It also might be less confusing to call the user by the UID we give it.

So perhaps:
```
groupadd -r 1337 -g 1337 && \
useradd -r -u 1337 -g 1337 1337 -d /app && \
```
?

Reply:
From there there downstream would switch `USER nodejs` for `USER 1337`.

Reply:
The problem with that is that it breaks implementations that have `USER nodejs`. I'd really like to keep the username until we know there's no downstreams.

Reply:
Happy to change the UID. I doubt 1337 is _a lot less_ likely to be set upstream though. I'd go for a non-obvious number.

Reply:
Normal users created without specifying the UID will increment starting from 1000 so 1337 should be fairly safe. (Though 1000 is probably reasonably safe and the build will break if it isn't so maybe we don't care?)

Perhaps we should make a minimal change for now (really just for documentation when you think about it) and think about changing things at a later date?

This PR could be made even safer by setting `-u 998` rather than `-u 1000` as that is the current UID that some projects _might_ already be referencing.

Reply:
@lennym and @daniel-ac-martin Isn't the point @KashifSaadat was making that the user needs to be changed to an integer, the user in the base image cannot be a string? Therefore, pick a number, set the user to that number in the base and publish a new major version of the base image. The only affected negatively downstream images will be those using :latest

Reply:
@JoeChapman: No. I think it would be more clear to do that (so we avoid having a name that we never reference). But the fix is for the downstream images to replace `USER nodejs` with `USER 998` (or `USER 1000` if we proceed with setting it to that).

i.e. We don't actually need to make a change here. (Though it is nice to for documentation and stability reasons.)

Reply:
@JoeChapman "Major versions" are tied to the node version, so "publish a new major version of the base image" isn't really a possibility.

Reply:
Also, for the record, most implementations that I'm aware of use the `v8` tag, which is not _exactly_ `latest`, but is equivalent since it's overwritten with each deployment. 

Hence my reticence to make breaking changes. They _will_ be breaking.